### DISCUSSION TITLE
Support for non-JSON responses

### DISCUSSION THREAD

**Discussion by humble-barnacle001** [JamesIves/fetch-api-data-action#468] [Ideas]
Could you please add support for responses which may be as text/plain by checking the response headers before parsing to JSON and saving the data in a txt file instead?

**Comment by JamesIves** [comment]
Seems reasonable. Do you have an API handy I can test this with?

**Reply by humble-barnacle001** [reply]
Here is link but headers are absent: https://hello-world-api-sbn2k60ruxvs.runkit.sh/

**Reply by humble-barnacle001** [reply]
Here is one with headers I created: https://fetch-api-data-action-test-scvrfk4481as.runkit.sh/

**Reply by JamesIves** [reply]
Thanks, I'll get supported added to this later next week.

**Comment by humble-barnacle001** [comment]
Thanks @JamesIves for the super quick response

**Reply by JamesIves** [reply]
I've got a test version of this feature ready if you'd like to give it a try. Instead of checking the headers the action will now only parse JSON if it's valid, otherwise it will save the raw response. You can modify the file format directly using the new `format` property, in your case you can set this to `txt` to generate a file of that type.

```yml
- name: Fetch API Data 📦
   uses: JamesIves/fetch-api-data-action@releases/v2
   with:
      endpoint: https://fetch-api-data-action-test-scvrfk4481as.runkit.sh
      debug: true
      format: txt
```

Let me know if this works for you! If it does I'll promote the build.

**Reply by JamesIves** [reply]
Released as part of `2.1.0`.

**Reply by humble-barnacle001** [reply]
> I've got a test version of this feature ready if you'd like to give it a try. Instead of checking the headers the action will now only parse JSON if it's valid, otherwise it will save the raw response. You can modify the file format directly using the new `format` property, in your case you can set this to `txt` to generate a file of that type.
> 
> ```yaml
> - name: Fetch API Data 📦
>    uses: JamesIves/fetch-api-data-action@releases/v2
>    with:
>       endpoint: https://fetch-api-data-action-test-scvrfk4481as.runkit.sh
>       debug: true
>       format: txt
> ```
> 
> Let me know if this works for you! If it does I'll promote the build.

Hey @JamesIves thanks for the release. I will surely check this out.

**Comment by humble-barnacle001** [comment]
I see after the release https://github.com/JamesIves/fetch-api-data-action/discussions/468#discussioncomment-1947657 and the release source in commit 570f45b8e098ffab8d1775253341cccb1c880eeb it is actually writing the data as text into the file encoded as utf-8.

Could you please also possibly allow an option to set the character encoding as desired?

**Reply by JamesIves** [reply]
Ah yeah- forgot about that bit. I can write a PR for it later today.

**Reply by humble-barnacle001** [reply]
Could I submit a PR?

**Reply by JamesIves** [reply]
Go for it! I'd start by searching for `format` and taking a look at how that property gets passed to the file generation function.

**Reply by humble-barnacle001** [reply]
Thanks for the help. I shall use it as my starting point.

**Reply by humble-barnacle001** [reply]
Hey I did some changes, and while running the tests, I get this error:
![image](https://user-images.githubusercontent.com/58736240/148985500-6f37522e-f583-4208-9b3a-2a6c81e6b3ce.png)

I guess this is due to different Node.js versions, isn't it?

**Reply by JamesIves** [reply]
Most likely yes. Your node version should match the one found in the nvmrc file. The contributing guide has details on how to see this up.

**Reply by JamesIves** [reply]
This has been published and released as part of v2.2.0. Thanks for your contribution!
